src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -99,18 +99,51 @@
 
     @Override
     public Connection connect(String url, Properties info) throws SQLException {
+        Pair<MongoConnection, Integer> p = getUnvalidatedConnectionAndTimeout(url, info);
+        Connection conn = p.left();
+        conn.isValid(p.right());
```

Comment:
Since you are making related changes, could you change ConnValidation.call return type to Void instead of Object?
This makes things totally clear this way that we don't expect a returned value.